tests/lib/devicetree/api/src/main.c

```diff
@@ -2667,6 +2667,70 @@ ZTEST(devicetree_api, test_mbox)
 				  DT_NODELABEL(test_mbox_zero_cell)), "");
 }
 
+ZTEST(devicetree_api, test_memory_attr)
+{
+	#define REGION_RAM_ATTR		(0xDEDE)
+	#define REGION_RAM_NOCACHE_ATTR	(0xCACA)
+
+	#define TEST_FUNC(p_name, p_base, p_size, p_attr)	\
+		{ .name = (p_name),				\
+		  .base = (p_base),				\
+		  .size = (p_size),				\
+		  .attr = (p_attr),				\
+		}
+
+	struct vnd_memory_binding {
+		char *name;
+		uintptr_t base;
+		size_t size;
+		unsigned int attr;
+	};
+
+	struct vnd_memory_binding val_apply[] = {
+		DT_MEMORY_ATTR_APPLY(TEST_FUNC)
+	};
+
+	zassert_true(!strcmp(val_apply[0].name, "memory@aabbccdd"), "");
+	zassert_equal(val_apply[0].base, 0xaabbccdd, "");
+	zassert_equal(val_apply[0].size, 0x4000, "");
+	zassert_equal(val_apply[0].attr, 0xDEDE, "");
+
+	zassert_true(!strcmp(val_apply[1].name, "memory@44332211"), "");
+	zassert_equal(val_apply[1].base, 0x44332211, "");
+	zassert_equal(val_apply[1].size, 0x2000, "");
+	zassert_equal(val_apply[1].attr, 0xCACA, "");
```

Comment:
Shouldn't this 0xCACA be `REGION_RAM_NOCACHE_ATTR`?

Reply:
yeah, maybe for readability but after all this is a test, so it doesn't really matter?